src/common/SplitView/HideSplitViewButton.tsx

```diff
@@ -0,0 +1,77 @@
+/**
+ * (c) 2022, Micro:bit Educational Foundation and contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+import { Box, Icon, IconButtonProps } from "@chakra-ui/react";
+import { RiDownloadLine } from "react-icons/ri";
+import CollapsibleButton from "../CollapsibleButton";
+import { splitViewHideButton } from "../zIndex";
+
+interface HideSplitViewButtonProps extends IconButtonProps {
+  handleClick: () => void;
+  direction: "left" | "right";
+  splitViewShown: boolean;
+  text?: string;
+}
+
+const HideSplitViewButton = ({
+  handleClick,
+  direction,
+  splitViewShown,
+  text = "",
+  ...props
+}: HideSplitViewButtonProps) => {
+  const mode = text ? "button" : "icon";
+  let rightBorderRadius = 6;
+  let leftBorderRadius = 0;
+  let rotation = "rotate(270deg)";
+  if (
+    (direction === "right" && splitViewShown) ||
+    (direction === "left" && !splitViewShown)
+  ) {
+    rightBorderRadius = 0;
+    leftBorderRadius = 6;
+    rotation = "rotate(90deg)";
+  }
+  return (
+    <Box position="relative">
+      {/* Hack to cover divider box shadow on right hand side. */}
+      {direction === "left" && splitViewShown && (
+        <Box
+          width="10px"
+          height="50px"
+          background="#eaecf1"
+          zIndex={5}
+          position="absolute"
+          left="-10px"
+          top="-5px"
+        />
+      )}
+      <CollapsibleButton
+        mode={text ? "button" : "icon"}
+        text={text}
+        icon={<Icon as={RiDownloadLine} transform={rotation} />}
+        fontSize="xl"
```

Comment:
lg to match the buttons below?

Otherwise, visually I really like this. Will look more closely at the styling tomorrow.